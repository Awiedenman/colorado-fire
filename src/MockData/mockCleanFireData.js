/* eslint-disable */
export const mockCleanFireData = [{
  fire_name: "416",
  acres_burned: " 54129 acres",
  image: undefined,
  last_update: " 7/24/2018 10:15:00 PM",
  latitude: "37.4605800000001",
  longitude: "-107.8081",
  verified: true
}]