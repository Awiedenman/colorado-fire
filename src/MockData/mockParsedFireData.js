export const mockParsedFireData = {
  rss: {
    channel: {
      item: [{
        description: {
          _text: " 7/24/2018 10:15:00 PM,  54129 acres, Lat/Lon: 37.4605800000001/-107.8081"
        },
          enclosure: {
            _attributes: {
              
            },
        },
          ['geo:lat']: {_text: "37.4605800000001"
        },
          ['geo:long']: {_text: "-107.8081"
        },
          link:{_text: "https://rmgsc-haws2.cr.usgs.gov/geomac/DynContent/georss/416 .png"
        },
          pubDate:{_text: "Wed, 25 Jul 2018 11:00:11 MDT"
        },
          title:{_text: "416 ,  2018-COSJF-000416"
        }
      }]
    }

  }
}