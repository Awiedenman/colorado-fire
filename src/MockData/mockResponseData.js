{
  _declaration: {
  _attributes: {
      version: "1.0"
    }
  },
  _instruction: {
    xml-stylesheet: "href='rssxsl.html' type='text/html' media='screen'"
  },
  rss: {
    "_attributes": {
      "version": "2.0",
      "xmlns:geo": "http://www.w3.org/2003/01/geo/wgs84_pos#",
      "xmlns:dc": "http://purl.org/dc/elements/1.1/"
    },
    "channel": {
      "title": {
        "_text": "NIFC Large Fires"
      },
      "description": {
        "_text": "Tue, 24 Jul 2018 11:30:10 MDT"
      },
      "link": {
        "_text": "http://www.nifc.gov/fireInfo/fireInfo_main.html"
      },
      "dc:publisher": {
        "_text": "U.S. Geological Survey"
      },
      "author": {
        "_text": "jlriegle@usgs.gov (Geosciences and Environmental Change Science Center)"
      },
      "pubDate": {
        "_text": "Tue, 24 Jul 2018 11:30:10 MDT"
      },
      "icon": {
        "_text": "D:\\webapps\\website\\geomac\\DynContent\\georss\\favicon.ico"
      },
      "item": [{
            "pubDate": {
              "_text": "Tue, 24 Jul 2018 11:30:10 MDT"
            },
            "title": {
              "_text": "Webber Creek,  2018-AKUYD-000135"
            },
            "description": {
              "_text": " 7/19/2018 10:30:00 PM,  6350.7 acres, Lat/Lon: 65.36033/-143.53233"
            },
            "link": {
              "_text": "https://rmgsc-haws1.cr.usgs.gov/geomac/DynContent/georss/Webber Creek.png"
            },
            "geo:lat": {
              "_text": "65.36033"
            },
            "geo:long": {
              "_text": "-143.53233"
            },
            "enclosure": {
              "_attributes": {
                "url": "Webber Creek.png"
              }
            }
          },
          {
            "pubDate": {
              "_text": "Tue, 24 Jul 2018 11:30:10 MDT"
            },
            "title": {
              "_text": "Zitziana River,  2018-AKTAD-000133"
            },
            "description": {
              "_text": " 7/19/2018 10:00:00 PM,  37524.6 acres, Lat/Lon: 64.7725/-150.72267"
            },
            "link": {
              "_text": "https://rmgsc-haws1.cr.usgs.gov/geomac/DynContent/georss/Zitziana River.png"
            },
            "geo:lat": {
              "_text": "64.7725"
            },
            "geo:long": {
              "_text": "-150.72267"
            },
            "enclosure": {
              "_attributes": {
                "url": "Zitziana River.png"
              }
            }
          },