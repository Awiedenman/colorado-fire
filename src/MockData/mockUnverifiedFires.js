 export const mockFiresFromDb = [{
   id: 10,
   first_name: "",
   last_name: "will's blaze",
   fire_name: null,
   image: null,
   acres_burned: null,
   last_update: null,
   latitude: "40",
   longitude: "-79",
   city: "",
   state: "",
   zip_code: "",
   verified: false,
   created_at: "2018-07-31T23:04:57.695Z",
   updated_at: "2018-07-31T23:04:57.695Z"
 }];