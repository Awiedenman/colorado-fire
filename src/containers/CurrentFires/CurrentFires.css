.gm-style-pbc {
  width: 40%;
}