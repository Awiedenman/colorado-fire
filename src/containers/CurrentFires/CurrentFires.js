import React, { Component } from 'react';
import { apiKey } from '../../api-key';
import { Map, InfoWindow, Marker, GoogleApiWrapper } from 'google-maps-react';
import { connect } from 'react-redux';


export class CurrentFires extends Component {
  
  render() {
    const currentFireMarkers = this.props.currentFires.map(fire => {
      return <Marker
          title={'The marker`s title will appear as a fire.'}
          name={fire.name}
          position={{lat: fire.latitude, lng: fire.longitude }}
          key={fire.name} 
        />
    })
      
    const initialCenter = {
      lat: 39,
      lng: -96.9
    }
    
    const style = {
      width: '100%',
      height: '100%'
    }
    
    return(
      <Map 
        style={style}
        google={this.props.google} 
        initialCenter={initialCenter}
        center={initialCenter}
        zoom= {5}
      >
      {currentFireMarkers}
        <InfoWindow onClose={this.onInfoWindowClose}>
            <div>
              {/* <h1>{this.state.selectedPlace.name}</h1> */}
            </div>
        </InfoWindow>
      </Map>
   )
  }
}

const mapStateToProps = (state) => ({
  currentFires: state.currentFires
})

const googleWrapper = GoogleApiWrapper({ apiKey: apiKey })(CurrentFires)

export default connect(mapStateToProps)(googleWrapper);

