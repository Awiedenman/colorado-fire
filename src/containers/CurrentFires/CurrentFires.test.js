import React from 'react';
import { shallow } from 'enzyme';
import { mockCleanFireData } from '../../MockData/mockCleanFireData';
import { CurrentFires, mapStateToProps } from './CurrentFires';

describe('CurrentFires', () => {

  describe('onMarkerClick', () => {

    test('should set the state ', () => {
      const wrapper = shallow(
        <CurrentFires 
          currentFires={mockCleanFireData}
        /> )
      // const mockCurrentFires = mockCleanFireData
      const mockMarker = {marker: 'A very nice marker'}
      const mockProps = {
        name: "416",
        image: undefined,
        acresBurned: " 54129 acres",
        lastUpdate: " 7/24/2018 10:15:00 PM",
        latitude: "37.4605800000001",
        longitude: "-107.8081"
      }

      wrapper.instance().onMarkerClick();

      expect(wrapper.state('selectedPlace')).toEqual(mockProps);
      expect(wrapper.state('activeMarker')).toEqual(mockMarker);
      expect(wrapper.state('showInfoWindow')).toEqual(true);

    })
  })

  describe('onMapClicked', () => {
    const wrapper = shallow(
        <CurrentFires 
          currentFires={mockCleanFireData}
        />)

    const mockProps = {};

    const mockNewState = {
      showingInfoWindow: false,
      activeMarker: null
    };

    wrapper.instance().onMapClicked();

    expect(wrapper.state('showInfoWindow')).toEqual(false);
    expect(wrapper.state('activeMarker')).toEqual(null);
  })

  describe('mapStateToProps', () => {
    test('should return an array with cleanedCurrentFireData from state', () => {
      const mockState = {
        currentFires: mockCleanFireData,
        test: 'testData'
      }
  
      const expected = {currentFires: mockCleanFireData};
  
      const mappedProps = mapStateToProps(mockState)
  
      expect(mappedProps).toEqual(expected)
    }) 
  })

  test('should match snapshot', () => {
    const wrapper = shallow(
      <CurrentFires
        currentFires={mockCleanFireData}
      />)

      expect(wrapper).toMatchSnapshot();
  })  
})