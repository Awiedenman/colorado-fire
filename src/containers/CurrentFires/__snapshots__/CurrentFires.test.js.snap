// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentFires should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentFires
    currentFires={
        Array [
            Object {
              "acresBurned": " 54129 acres",
              "image": undefined,
              "lastUpdate": " 7/24/2018 10:15:00 PM",
              "latitude": "37.4605800000001",
              "longitude": "-107.8081",
              "name": "416",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Map
        center={
                Object {
                        "lat": undefined,
                        "lng": undefined,
                      }
        }
        centerAroundCurrentLocation={false}
        containerStyle={Object {}}
        google={undefined}
        initialCenter={
                Object {
                        "lat": 45,
                        "lng": -105,
                      }
        }
        maxZoom={14}
        minZoom={3.4}
        onClick={[Function]}
        style={Object {}}
        visible={true}
        zoom={3.4}
>
        <Marker
                acresBurned=" 54129 acres"
                google={undefined}
                lastUpdate=" 7/24/2018 10:15:00 PM"
                name="<a href='https://www.google.com/search?416>fire.name</a>"
                onClick={[Function]}
                position={
                        Object {
                                "lat": "37.4605800000001",
                                "lng": "-107.8081",
                              }
                }
                title="416"
        />
        <InfoWindow
                marker={Object {}}
                visible={false}
        >
                <div>
                        <h1 />
                        <h3>
                                Current Burn: 
                        </h3>
                        <h3>
                                Last updated on 
                        </h3>
                </div>
        </InfoWindow>
</Map>,
      "className": "map",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "center": Object {
          "lat": undefined,
          "lng": undefined,
        },
        "centerAroundCurrentLocation": false,
        "children": Array [
          Array [
            <Marker
              acresBurned=" 54129 acres"
              google={undefined}
              lastUpdate=" 7/24/2018 10:15:00 PM"
              name="<a href='https://www.google.com/search?416>fire.name</a>"
              onClick={[Function]}
              position={
                            Object {
                                          "lat": "37.4605800000001",
                                          "lng": "-107.8081",
                                        }
              }
              title="416"
/>,
          ],
          <InfoWindow
            marker={Object {}}
            visible={false}
>
            <div>
                        <h1 />
                        <h3>
                                    Current Burn: 
                        </h3>
                        <h3>
                                    Last updated on 
                        </h3>
            </div>
</InfoWindow>,
        ],
        "containerStyle": Object {},
        "google": undefined,
        "initialCenter": Object {
          "lat": 45,
          "lng": -105,
        },
        "maxZoom": 14,
        "minZoom": 3.4,
        "onClick": [Function],
        "style": Object {},
        "visible": true,
        "zoom": 3.4,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "416",
          "nodeType": "class",
          "props": Object {
            "acresBurned": " 54129 acres",
            "google": undefined,
            "lastUpdate": " 7/24/2018 10:15:00 PM",
            "name": "<a href='https://www.google.com/search?416>fire.name</a>",
            "onClick": [Function],
            "position": Object {
              "lat": "37.4605800000001",
              "lng": "-107.8081",
            },
            "title": "416",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              <h1 />
              <h3>
                            Current Burn: 
              </h3>
              <h3>
                            Last updated on 
              </h3>
</div>,
            "marker": Object {},
            "visible": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1 />,
                <h3>
                  Current Burn: 
</h3>,
                <h3>
                  Last updated on 
</h3>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Current Burn: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Current Burn: ",
                  undefined,
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Last updated on ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Last updated on ",
                  undefined,
                ],
                "type": "h3",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Map
          center={
                    Object {
                              "lat": undefined,
                              "lng": undefined,
                            }
          }
          centerAroundCurrentLocation={false}
          containerStyle={Object {}}
          google={undefined}
          initialCenter={
                    Object {
                              "lat": 45,
                              "lng": -105,
                            }
          }
          maxZoom={14}
          minZoom={3.4}
          onClick={[Function]}
          style={Object {}}
          visible={true}
          zoom={3.4}
>
          <Marker
                    acresBurned=" 54129 acres"
                    google={undefined}
                    lastUpdate=" 7/24/2018 10:15:00 PM"
                    name="<a href='https://www.google.com/search?416>fire.name</a>"
                    onClick={[Function]}
                    position={
                              Object {
                                        "lat": "37.4605800000001",
                                        "lng": "-107.8081",
                                      }
                    }
                    title="416"
          />
          <InfoWindow
                    marker={Object {}}
                    visible={false}
          >
                    <div>
                              <h1 />
                              <h3>
                                        Current Burn: 
                              </h3>
                              <h3>
                                        Last updated on 
                              </h3>
                    </div>
          </InfoWindow>
</Map>,
        "className": "map",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "center": Object {
            "lat": undefined,
            "lng": undefined,
          },
          "centerAroundCurrentLocation": false,
          "children": Array [
            Array [
              <Marker
                acresBurned=" 54129 acres"
                google={undefined}
                lastUpdate=" 7/24/2018 10:15:00 PM"
                name="<a href='https://www.google.com/search?416>fire.name</a>"
                onClick={[Function]}
                position={
                                Object {
                                                "lat": "37.4605800000001",
                                                "lng": "-107.8081",
                                              }
                }
                title="416"
/>,
            ],
            <InfoWindow
              marker={Object {}}
              visible={false}
>
              <div>
                            <h1 />
                            <h3>
                                          Current Burn: 
                            </h3>
                            <h3>
                                          Last updated on 
                            </h3>
              </div>
</InfoWindow>,
          ],
          "containerStyle": Object {},
          "google": undefined,
          "initialCenter": Object {
            "lat": 45,
            "lng": -105,
          },
          "maxZoom": 14,
          "minZoom": 3.4,
          "onClick": [Function],
          "style": Object {},
          "visible": true,
          "zoom": 3.4,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "416",
            "nodeType": "class",
            "props": Object {
              "acresBurned": " 54129 acres",
              "google": undefined,
              "lastUpdate": " 7/24/2018 10:15:00 PM",
              "name": "<a href='https://www.google.com/search?416>fire.name</a>",
              "onClick": [Function],
              "position": Object {
                "lat": "37.4605800000001",
                "lng": "-107.8081",
              },
              "title": "416",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                <h1 />
                <h3>
                                Current Burn: 
                </h3>
                <h3>
                                Last updated on 
                </h3>
</div>,
              "marker": Object {},
              "visible": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1 />,
                  <h3>
                    Current Burn: 
</h3>,
                  <h3>
                    Last updated on 
</h3>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Current Burn: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Current Burn: ",
                    undefined,
                  ],
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Last updated on ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Last updated on ",
                    undefined,
                  ],
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
